rfcs/0013-transition-caching.md

```diff
@@ -0,0 +1,40 @@
+# Transition Caching
+
+## Summary
+[summary]: #summary
+
+A new transition caching logic for the transition router which aims to track transitions which are still being processed as well as transitions which have failed validation.
+
+## Motivation
+[motivation]: #motivation
+
+Within the transition router system, the only check for duplicate transitions is performed by the transition validator, and each transition is only checked against the transitions which are currently in the transition frontier. However, there are two types of duplicate transitions which are not being checked for: transitions which are still being processed by the system (either in the processor pipe or in the catchup scheduler and catchup thread), and transitions which have been determined to be invalid. In the case of the former, the system ends up processing more transitions than necessary, and the number of duplicated processing increases along with the networks size. In the case of the latter, the system is opened up for DDoS attacks since an adversary could continously send transitions with valid proofs but invalid staged ledger diffs, causing each node to spend a significant enough amount of time before invalidating the transition each time it recieves it.
```

Comment:
Can you elaborate on a case where a user sends invalid staged_ledger_diffs, but valid proofs. Is there a way to see that diff does not correspond with the proof?

Reply:
I'm not if I understand what you mean. Staged ledger diffs are not proved in the snark, they require separate validation, which can only be done by applying them to the previous state in the blockchain.